internals/types/index.d.ts

```diff
@@ -26,13 +26,38 @@ export interface JSONObject {
   [key: string]: JSONValue;
 }
 
-export interface AuthConfig {
+interface AuthConfigBase {
   '// Note'?: string;
   '// Docs'?: string;
   token?: string;
   skipWrite?: boolean;
 }
 
+interface LegacyAuthConfig extends AuthConfigBase {
+  /** An Vercel token retrieved using the legacy authentication flow. */
+  token?: string;
+  type?: 'legacy';
+}
+
+interface OAuthAuthConfig extends AuthConfigBase {
+  /** An `access_token` obtained using the OAuth Device Authorization flow.  */
+  token?: string;
+  /**
+   * The absolute time (seconds) when the {@link OAuthAuthConfig.token} expires.
+   * Used to optimistically check if the token is still valid.
+   */
+  expiresAt?: number;
+  refreshToken?: string;
+  /**
+   * The absolute time (seconds) when the {@link OAuthAuthConfig.refreshToken} expires.
+   * Used to optimistically check if the token is still valid.
+   */
+  refreshTokenExpiresAt?: number;
```

Comment:
IMO, we do not even need to track this. If we have a refresh token, we can try to use it, irrespective of its expiry.